### ISSUE TITLE
How to create an Alexa Skill in new Developer Console

### ISSUE THREAD

**Issue by cigdangle** [stevenleeg/geemusic#277]
I had GeeMusic working at one time, but am no longer able to get it working.  The issue is I am unsure how to setup an Alexa skill in the new Alexa Developer Console.  There are a number of additional options, and I'm too unfamiliar with what they are asking.

HELP!

**Comment by fergyfresh** [comment]
You're going to have to be WAYYYY more specific.

**Comment by cigdangle** [comment]
The directions do not apply to the current Alexa Developer Console, so you cannot follow them to create a skill.  Here is a small snippet of what is needed to create a skill now:

![image](https://user-images.githubusercontent.com/31552562/49891999-8689ae00-fe0d-11e8-83b1-e8492bee9881.png)

**Comment by fergyfresh** [comment]
LOL. It seems like they change the interface, but the big pieces are still in there.

**Comment by cigdangle** [comment]
Yes, I think so too, but it's just confusing me.  For example. when you click #2 above, you see this:
![image](https://user-images.githubusercontent.com/31552562/49961045-442ca380-fed7-11e8-870b-153ac05ffc86.png)

**Comment by cigdangle** [comment]
Yes, I think so too, but it's just confusing me. For example. when you click #2 above, you see this:
![image](https://user-images.githubusercontent.com/31552562/49961087-61617200-fed7-11e8-970d-3b6727c459b1.png)

**Comment by RandyCupic** [comment]
@cigdangle Yes, it's a little bit different and I had some difficulties as well, but carefully following the instructions and looking for similar options, I've managed to get it working. For example, the last image you posted: the instructions contain a table  which says:
Skill Type | Custom Interaction Model
-- | --

So this is telling you that you should select the `Create custom intent` option on your image. If you will have any more difficulties, just ask and I can try to help you since I got it working on the "new" console a week ago :)

**Comment by fergyfresh** [comment]
If anyone notices any changes with the setup feel free to update the README. I'm looking at you @RandyCupic and @cigdangle. I set it up once like 18 months ago and haven't had to change it since. I just push code updates to my Lambda function.

**Comment by RandyCupic** [comment]
@fergyfresh I hope you don't mind if I use this issue to ask a question what's been bothering me: for what are we using the S3 bucket? I don't really understand this whole concept over AWS but I noticed that it generates a lot of traffic and it started pumping my AWS bills. There are around 8 GBs already of data, but I don't think that I've been listening to such a huge amount of songs.

What is the S3 bucket used for and can I somehow prevent the traffic to go over AWS? Can't we just feed Alexa with direct streaming URL on Google servers and stream from there? Or is there some kind of API limit so that we're forced to cache songs?

**Comment by RandyCupic** [comment]
Well 100-300 songs is nothing :) I listened to about 50 songs only yesterday. But the price is different than yours. They charged me 0.16 $ for these 50 songs. Even though, 50 songs I have scrobbled in last.fm, in reality, there's probably at least 30% more, since I like to skip songs a lot.

But how do you get only 0.11 GB for 300 songs? When I open my S3 bucket, I can see tons of entries there. I guess that each entry is a song, right? And lots of them have over 10 MB. So 100 songs = approx. 1 GB. Let's say that I listen to at least 1000 songs monthly -> that's 10 GB * 0.09 $ + VAT -> at least 1$ monthly. But I'm already at 0.56$ for this month.

Here's my bill:
![image](https://user-images.githubusercontent.com/7003823/50041752-28053f80-005a-11e9-881a-786edab1686d.png)

I was reading a little bit about S3 buckets on Google and I've found out that bad settings with servers and zones could cause bigger charges. I'm from Croatia, and my Echo Dot is set to Seattle, because Amazon doesn't support Croatia, and so that I can use Pandora which is not supported in Croatia.

**Comment by fergyfresh** [comment]
Yeah. You don't have to use S3. It's actually an environment variable. `USE_S3_BUCKET`. You could add another `proxy_response` implementation that uses your Rasperry PI as a proxy instead of S3 (if you wanna still use Lambda) Or you could just deploy this to your server in house which is what I think a majority of users here do. I am the black sheep here that uses AWS Lambda + S3 + API Gateway.

The function you'd have to implement if you ONLY wanna move from S3 is here https://github.com/stevenleeg/geemusic/blob/master/geemusic/controllers.py#L12

**Comment by fergyfresh** [comment]
Does that make sense? @RandyCupic.

**Comment by RandyCupic** [comment]
Yes, thank you :) I just wanted to check before moving to the local server, if there's anything I could do. Lambda will do for now, until I get some time to set-up my local server.

**Comment by cigdangle** [comment]
Can we get back to my question now? :)

I really need some step-by-step guidance setting up the Alexa Skill.  Help!

**Comment by fergyfresh** [comment]
Try and try again. You'll learn enough. Honestly everything in that screenshot is what I made the current README off of. Many others have successfully done it from this readme. You have to try a little. I am busy, I'm sorry.

**Comment by ev0ldave** [comment]
@cigdangle cigdangle, I just finished this
1. Invocation name, gee music
2. intents, samples, slots:  left panel under Slot Types is JSON Editor, drag and drop the json file at speech_assets/interactionModel.json
3. Interfaces at left panel, turn on Audio Player
4. Endpoint, set as described in the current README as well as the ssl setup in drop down
5. Click on the Build in the top bar then click on the 3. Build Model on the right panel (This will build your skill)

Once the build is done click on the "test" in the top bar. enable testing in development.  If using Heroku I had to explicitly call out the full URL instead of letting HEROKU_APP_NAME resolve, it would not work.. You will see all of this output in the JSON Output window to see if URLs are building properly.  Once tested I had to split from the current README and fill out Distribution section, Privacy & Compliance.  I had to do this in order to enable beta test under the Availability Tab there and get it onto my echo.  I added myself to the beta test list and copy/pasted into new browser window the invite URL.
